### DISCUSSION TITLE
Using PIR in combination with a pushbutton

### DISCUSSION THREAD

**Discussion by Baptiste-mrch** [arendst/Tasmota#16228] [Q&A] [answered]
Hello,

I have a project with Tasmota that turn on or off a LED. I have 2 inputs : 
- One pushbutton that toggle the light
- One PIR sensor that turn on the light for a brief moment
The behavior I want to implement is that if the light is turned off, the PIR can turn on the light and off after a brief moment. BUT if the light is already turned on (using the pushbutton or the interface), then I want the PIR to do nothing of the light.
A little schema to explain what I wanted to implement :
![image](https://user-images.githubusercontent.com/62337736/184701929-cf5a0299-f565-423a-80e5-05b864f859c2.png)

Can someone help me with that?
Thanks a lot 😄

**Comment by barbudor** [comment, marked as answer]
You need to use [Rules](https://tasmota.github.io/docs/Rules/) and carefully read about the [Switchmodes](https://tasmota.github.io/docs/Buttons-and-Switches/#switchmode)

The button should be connected to a GPIO configured as Button_1 (or Button_i1 if inverted) and the behavior is already what you are expecting.
The PIR should be connected to a GPIO configured as Switch_1 using Switchmode 13 (or 14 if inverted) so that only the ON is generating an event.
Using a var to store the information that the power is already on or off(so the PIR has no effect if the power is already on)
Using a RuleTimer for your temporary light on (assuming 10 seconds for below)

This should provide you what you are looking for:
```
Rule1
  on power#state=0 do backlog var1 1; ruletimer1 0 endon
  on power#state=1 do var1 0 endon
  on switch1#state=%var1% do backlog power on; ruletimer1 10 endon
  on rules#timer=1 do power off
Rule1 1
```
The Rule1 command should be entered as a single line. It is split here for readability

**Reply by Baptiste-mrch** [reply]
Thanks for your reply.
Is it possible to keep the timer of the PIR? So that Tamota will listen when the PIR is turning off and then turning off the light.
If not, your rule works fine, but the light doesn't turn off after the 10 seconds. Do you have any ideas why?

**Reply by Baptiste-mrch** [reply]
I put uppercase on the commands, it now works:
```
Rule1
  ON Power1#state=0 do backlog var1 1; RuleTimer1 0 ENDON
  ON Power1#state=1 do var1 0 ENDON
  ON switch1#state=%var1% do backlog Power1 on; RuleTimer1 10 ENDON
  ON Rules#Timer=1 do Power1 off ENDON
```
Thanks a lot for your help

**Reply by barbudor** [reply]
The rules are not case sensitive, so uppercase would not impact the behavior
Not sure to understand the problem in the previous message. Is that ok now ?

**Reply by sfromis** [reply]
There was an important misstep in:
`on power#state=0 do backlog var1 1; ruletimer1 0 endon`
While the `Powerx` *commands* defaults to channel 1, that is not the case for the trigger names.

**Reply by barbudor** [reply]
😨
100% correct @sfromis 
Thanks
